### PR TITLE
Base api improvements

### PR DESCRIPTION
**Pull request by nishadmusthafa**

### CODE REVIEW

File: gaelib/tests/views/test_base_view.py [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+  @staticmethod
+  def get_entities(entity_id=None):
+    kwargs = {}
+    if entity_id:
+      kwargs = {'key_strs': [entity_id]}
+    return SampleModel.retrieve(**kwargs)
+
+  @staticmethod
+  def validate_fields(**kwargs):
+    # perform validation here
```
Comment (shantanumallik) [inline gaelib/tests/views/test_base_view.py:41]: is this a placeholder for now?

File: gaelib/utils/web.py [hunk 2]
```diff
@@ -63,58 +79,53 @@ def log_request_info():
...
-    # if get_app_or_default_prop('DATASTORE_PROJECT_ID'):
-    #   os.environ['DATASTORE_PROJECT_ID'] = get_app_or_default_prop('DATASTORE_PROJECT_ID')
-    # if get_app_or_default_prop('GOOGLE_CLOUD_PROJECT'):
-    #   os.environ['GOOGLE_CLOUD_PROJECT'] = get_app_or_default_prop('GOOGLE_CLOUD_PROJECT')
-
-    # Dumb hack we need to run locally
+    # Dumb hack we need to run the app locally
     # The private key that's in this file is not associated with
     #  a real account
     os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = os.getcwd() + \
```
Comment (shantanumallik) [inline gaelib/utils/web.py:98]: why don't we add the environment var for DATASTORE_EMULATOR_HOST, DATASTORE_PROJECT_ID AND GOOGLE_CLOUD PROJECT? Those are necessary for running the app in the emulator


    os.environ['DATASTORE_EMULATOR_HOST'] = 'localhost:8089'
    os.environ['DATASTORE_PROJECT_ID'] = <any project id>
    os.environ['GOOGLE_CLOUD_PROJECT'] = <same project id as above>

File: gaelib/view/base_view.py [hunk 3]
```diff
@@ -84,12 +91,71 @@ def validate_request_args(self):
...
-class BaseApiHandler(BaseHttpHandler):
+class BaseAPIHandler(BaseHttpHandler):
+  controller = None
   decorators = [auth_required]
 
+  def validate(self, entity_id):
+    if not self.controller:
+      error = "No controller defined"
+      response, _, _ = self.json_error(error, 500)
+      raise HTTPException(error, response)
```
Comment (shantanumallik) [inline gaelib/view/base_view.py:105]: HTTPException expects the return to be of type `Response | None`. Here we are returning a response dict. This did raise an error during testing with the package in cm-facade-jira

File: gaelib/tests/views/test_base_view.py (hunk 1 above)
Comment (nishadmusthafa) [reply gaelib/tests/views/test_base_view.py:41]: that's correct . something can can be added later.

File: gaelib/utils/web.py (hunk 2 above)
Comment (nishadmusthafa) [reply gaelib/utils/web.py:98]: We've set it up as part of the docker run command in run_test_container.sh in gaelib or run_dev_container.sh in the app repositories.

File: gaelib/view/base_view.py (hunk 3 above)
Comment (nishadmusthafa) [reply gaelib/view/base_view.py:105]: From external chat, it was learnt that this was the error
>Argument of type “tuple[Response, Unknown, dict[Unknown, Unknown]]” cannot be assigned to parameter “response” of type “Response | None” in function “__init__”

Comment (nishadmusthafa) [reply gaelib/view/base_view.py:105]: I've decided to take up all lint related fixes in another pull request. So punting this to then.

**Pull request merged**